sdk/src/Services/DynamoDBv2/Custom/DataModel/TransactWrite.cs

```diff
@@ -437,6 +435,43 @@ private Expression CreateConditionExpressionForVersion(ItemStorage storage)
                 DocumentTransaction.TargetTable.IsEmptyStringValueEnabled);
             return DynamoDBContext.CreateConditionExpressionForVersion(storage, conversionConfig);
         }
+        
+
+        private void AddDocumentTransaction(ItemStorage storage, Expression conditionExpression)
+        {
+            var hashKeyPropertyNames = storage.Config.HashKeyPropertyNames;
+            var rangeKeyPropertyNames = storage.Config.RangeKeyPropertyNames;
+
+            var keyList = storage.Document.Keys.ToList();
```

Comment:
Can you name the variable `attributeNames`? My first scan of the PR got confused thinking `storage.Document.Keys` was returning the DynamoDB key names. I should have known better but maybe seeing the variable as `attributeNames` instead of `keyList` would help the next reader not jump to that collection. Ideally we would have named `Document.Keys` something better but we can't do anything about that now.